Invalid use of `AC_CHECK_PROGS`

The template for `AC_CHECK_PROGS` is:
```
AC_CHECK_PROGS (variable, progs-to-check-for, [value-if-not-found], [path = ‘$PATH’])
```

probably confused with:
```
AC_CHECK_PROG (variable, prog-to-check-for, value-if-found, [value-if-not-found], [path = ‘$PATH’], [reject])
```

which takes `value-if-found` as its third parameter.

`AC_CHECK_PROGS` instead sets the variable directly to the program we are looking for, in which case `curl(1)`.

@zvr Trying to run the benchmark would result to errors, so I looked around and came upon the GNU doc for autoconf:
https://www.gnu.org/software/autoconf/manual/autoconf-2.68/html_node/Generic-Programs.html

which states:

>  — Macro: AC_CHECK_PROG (variable, prog-to-check-for, value-if-found, [value-if-not-found], [path = ‘$PATH’], [reject])
> 
>    Check whether program prog-to-check-for exists in path. If it is found, set variable to value-if-found, otherwise to
value-if-not-found, if given. Always pass over reject (an absolute file name) even if it is the first found in the search path; 
in that case, set variable using the absolute file name of the prog-to-check-for found that is not reject. If variable was
 already set, do nothing. Calls AC_SUBST for variable. The result of this test can be overridden by setting the variable 
variable or the cache variable ac_cv_prog_variable. 
> 
> — Macro: AC_CHECK_PROGS (variable, progs-to-check-for, [value-if-not-found], [path = ‘$PATH’])
> 
>    Check for each program in the blank-separated list progs-to-check-for existing in the path. If one is found, set variable
> to the name of that program. Otherwise, continue checking the next program in the list. If none of the programs in the list
>   are found, set variable to value-if-not-found; if value-if-not-found is not specified, the value of variable is not changed. 
>   Calls AC_SUBST for variable. The result of this test can be overridden by setting the variable variable or the cache variable
>    ac_cv_prog_variable. 

Which validates my point above. And also explains my error, which was:

> curl https://ftp.gnu.org/gnu/hello/hello-2.12.1.tar.gz
> Warning: Binary output can mess up your terminal. Use "--output -" to tell 
> Warning: curl to output it to your terminal anyway, or consider "--output 
> Warning: <FILE>" to save to a file.
> make: *** [Makefile:413: hello-2.12.1.tar.gz] Error 23

Showing that `curl` was used for `DOWNLOAD` Instead of `curl -O`.

--- a/benchmark/Makefile.am
+++ b/benchmark/Makefile.am
@@ -15,6 +15,8 @@ endif
 
 if !HAVE_TIME
 $(error Did not find "time" in the PATH.)
+else
+TIMER = $(TIME) -v
 endif
 
 
@@ -33,9 +35,9 @@ $(LOCAL_DIR)/config.h: $(LOCAL_DIR)
 
 $(BENCHOUT): $(LOCAL_DIR)/config.h
 	cd $(LOCAL_DIR) && make clean
-	cd $(LOCAL_DIR) && $(TIME) make
+	cd $(LOCAL_DIR) && $(TIMER) make
 	cd $(LOCAL_DIR) && make clean
-	cd $(LOCAL_DIR) && $(TIME) ../$(EXE) make
+	cd $(LOCAL_DIR) && $(TIMER) ../$(EXE) make
 
 compile_single_file: f1.c
 	$(EXE) -o $@.out $(CC) -c $^

--- a/configure.ac
+++ b/configure.ac
@@ -26,12 +26,12 @@ dnl Where are the sources
 AC_CONFIG_SRCDIR([src/main.c])
 
 dnl Checks for programs
-AC_CHECK_PROGS(DOWNLOAD, curl, [curl -O])
+AC_CHECK_PROG(DOWNLOAD, curl, [curl -O])
 if test "x$DOWNLOAD" = x; then
-    AC_CHECK_PROGS(DOWNLOAD, wget, wget)
+    AC_CHECK_PROG(DOWNLOAD, wget, wget)
 fi
 AM_CONDITIONAL([HAVE_DOWNLOAD], [test -n "$DOWNLOAD"])
-AC_CHECK_PROGS(TIME, time, time -v)
+AC_PATH_PROG(TIME, time)
 AM_CONDITIONAL([HAVE_TIME], [test -n "$TIME"])
 
 dnl Checks for libraries

